src/main/kotlin/no/kartverket/matrikkel/bygning/services/EgenregistreringService.kt

```diff
@@ -1,93 +1,50 @@
 package no.kartverket.matrikkel.bygning.services
 
 import no.kartverket.matrikkel.bygning.matrikkel.BygningClient
-import no.kartverket.matrikkel.bygning.models.BruksenhetRegistrering
 import no.kartverket.matrikkel.bygning.models.Bygning
-import no.kartverket.matrikkel.bygning.models.BygningRegistrering
+import no.kartverket.matrikkel.bygning.models.Egenregistrering
 import no.kartverket.matrikkel.bygning.models.Result
 import no.kartverket.matrikkel.bygning.models.responses.ErrorDetail
-import no.kartverket.matrikkel.bygning.routes.v1.dto.request.EgenregistreringRequest
-import java.time.Instant
-import java.util.*
+import no.kartverket.matrikkel.bygning.repositories.EgenregistreringRepository
 
-class EgenregistreringService(private val bygningClient: BygningClient) {
-    private val bygningRegistreringer: MutableList<BygningRegistrering> = mutableListOf()
-    private val bruksenhetRegistreringer: MutableList<BruksenhetRegistrering> = mutableListOf()
+class EgenregistreringService(
+    private val bygningClient: BygningClient, private val egenregistreringRepository: EgenregistreringRepository
+) {
+    fun addEgenregistrering(egenregistrering: Egenregistrering): Result<Int> {
 
-    fun addEgenregistreringToBygning(egenregistrering: EgenregistreringRequest): Result<Unit> {
-        val bygningId = egenregistrering.bygningId
-
-        val bygning = bygningClient.getBygningById(bygningId) ?: return Result.ErrorResult(
+        val bygning = bygningClient.getBygningById(egenregistrering.bygningId) ?: return Result.ErrorResult(
             ErrorDetail(
-                detail = "Bygningen finnes ikke i matrikkelen",
+                detail = "Bygning med ID ${egenregistrering.bygningId} finnes ikke i matrikkelen",
             ),
         )
 
-        if (!isAllBruksenheterRegisteredOnCorrectBygning(egenregistrering, bygning)) {
+        val invalidBruksenheter = findBruksenheterNotRegisteredOnCorrectBygning(egenregistrering, bygning)
+        if (invalidBruksenheter.isNotEmpty()) {
             return Result.ErrorResult(
                 ErrorDetail(
-                    detail = "Bruksenheten finnes ikke i bygningen",
+                    detail = "Bruksenhet${if (invalidBruksenheter.size > 1) "er" else ""} med ID ${invalidBruksenheter.joinToString()} finnes ikke i bygning med ID ${bygning.bygningId}",
                 ),
             )
         }
 
-        saveEgenregistreringToBygning(egenregistrering)
-        saveEgenregistreringToBruksenhet(egenregistrering)
-        return Result.Success(Unit)
-    }
-
-    private fun isAllBruksenheterRegisteredOnCorrectBygning(
-        egenregistrering: EgenregistreringRequest, bygning: Bygning
-    ): Boolean {
-        if (egenregistrering.bruksenhetRegistreringer?.isEmpty() == true) return true
+        val egenregistreringSaveResult = egenregistreringRepository.saveEgenregistrering(egenregistrering);
```

Comment:
Foreslår å bare inline:
```suggestion
        return egenregistreringRepository.saveEgenregistrering(egenregistrering);
```